### PR TITLE
Suppress `opaque_hidden_inferred_bound` for nested RPITs

### PR DESCRIPTION
**Pull request by compiler-errors**
They trigger too much, making repos like linkerd/linkerd2-proxy#2275 sad.

Ideally, at least for RPITs (and probably TAITs?), specifically when we have `impl Trait<Assoc = impl ..>`, that nested opaque should have the necessary `Assoc` item bounds elaborated into its own item bounds. But that's another story.

r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
idk... at that point the lint is pretty much useless. Not sure what the right way forward is...

**Comment by oli-obk** [comment]
Thought some more about this. I guess for nested rpits there's a good path forward, so let's not annoy users about it

@bors r+ rollup

File: compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use rustc_hir as hir;
 use rustc_infer::infer::TyCtxtInferExt;
 use rustc_macros::{LintDiagnostic, Subdiagnostic};
 use rustc_middle::ty::{
-    self, fold::BottomUpFolder, print::TraitPredPrintModifiersAndPath, Ty, TypeFoldable,
+    self, fold::BottomUpFolder, print::TraitPredPrintModifiersAndPath, DefIdTree, Ty, TypeFoldable,
```
Comment (Nilstrieb) [inline compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs:5]: Needs to be rebased over #108351

**Pull request merged**